### PR TITLE
Donate

### PR DESCRIPTION
**Pull request by dimka14101**

### CODE REVIEW

File: FundTrack.WebUI/ClientApp/app/services/concrete/finance/donate-money.service.ts [hunk 1]
```diff
@@ -34,20 +34,15 @@ export class DonateService {
     }
 
     getAccountsForDonate(organizationId: number): Observable<OrganizationDonateAccountsViewModel> {
-        return this._http.get('api/Donate/GetAccountsForDonate/' + organizationId.toString()).
+        return this._http.get('api/Donate/AccountsForDonate/' + organizationId.toString()).
```
Comment (Barabashka88) [inline FundTrack.WebUI/ClientApp/app/services/concrete/finance/donate-money.service.ts]: move url to separate file

File: FundTrack.WebUI/ClientApp/app/components/finance/donate-money.component.ts [hunk 2]
```diff
@@ -178,21 +178,17 @@ export class MakeDonationComponent implements OnInit, DoCheck, OnDestroy{
...
         this.fondyPayModel.signature = this.createSignature(this.fondyPayModel);
-
-        if (this.target.targetId == undefined) {
-            this.showErrorsForSelect();
-            return;
-        }
-
-        this.merchantData.targetId = this.target.targetId.toString();  
+      
+        this.merchantData.targetId = this.accountForDonation.targetId==null ? "5":this.accountForDonation.targetId.toString();
```
Comment (Barabashka88) [inline FundTrack.WebUI/ClientApp/app/components/finance/donate-money.component.ts]: why 5???

File: FundTrack.BLL/Concrete/DonateMoneyService.cs [hunk 3]
```diff
@@ -18,7 +20,8 @@ public DonateMoneyService(IUnitOfWork unitOfWork)
 
         public OrganizationDonateAccountsViewModel GetAccountForDonation(int organizationId)
         {
-            var orgAccounts = _unitOfWork.OrganizationAccountRepository.ReadOrgAccountsForDonations(organizationId);
+            var orgAccounts = _unitOfWork.OrganizationAccountRepository.ReadOrgAccountsForDonations(organizationId).Distinct((c1,c2)=> c1.Target == c2.Target);
```
Comment (unclefrost) [inline FundTrack.BLL/Concrete/DonateMoneyService.cs]: formatting

File: FundTrack.BLL/Concrete/DonateMoneyService.cs [hunk 4]
```diff
@@ -33,10 +36,13 @@ public OrganizationDonateAccountsViewModel GetAccountForDonation(int organizatio
...
                         Description = orgAccount.Description,
                         BankAccountId = orgAccount.BankAccount.Id,
-                        MerchantId = (int)orgAccount.BankAccount.MerchantId,
+                        MerchantId = orgAccount.BankAccount.MerchantId,
                         MerchantPassword = orgAccount.BankAccount.MerchantPassword,
-                        Name = orgAccount.OrgAccountName
-                    });
+                        Name = orgAccount.OrgAccountName,
+                        TargetId = orgAccount.TargetId,
+                        Target = orgAccount.TargetId == null ? "Загальний" : _unitOfWork.TargetRepository.GetTargetById((int)orgAccount.TargetId).TargetName
```
Comment (unclefrost) [inline FundTrack.BLL/Concrete/DonateMoneyService.cs]: assign to a variable (move to a separate line)

Comment (unclefrost) [reply FundTrack.BLL/Concrete/DonateMoneyService.cs]: (int)orgAccount.TargetId - use .GetValueOrDefault();

File: FundTrack.WebUI/Controllers/DonateController.cs [hunk 5]
```diff
@@ -70,24 +70,18 @@ public string CheckPayment([FromBody] CheckStatusRequestViewModel request)
...
         /// <returns>Organization id and accounts</returns>
-        [HttpGet("GetAccountsForDonate/{organizationId}")]
+        [HttpGet("AccountsForDonate/{organizationId}")]
         public OrganizationDonateAccountsViewModel GetAccountsForDonate(int organizationId)
         {
             return _donateMoneyService.GetAccountForDonation(organizationId);
         }
 
-        [HttpGet("GetOrderId")]
+        [HttpGet("OrderId")]
```
Comment (unclefrost) [inline FundTrack.WebUI/Controllers/DonateController.cs]: "order/id"

Comment (dimka14101) [reply FundTrack.WebUI/Controllers/DonateController.cs]: this method does not accept any argument, but serves to obtain a unique identifier for a financial transaction so /{id} is unnecessary

**Pull request merged**